astroquery/esa_hubble/core.py

```diff
@@ -0,0 +1,374 @@
+            flag to display information about the process
+
+        Returns
+        -------
+        A Job object
+        """
+
+        return self._tap.launch_job(query=query, output_file=output_file,
+                                    output_format=output_format,
+                                    verbose=False,
+                                    dump_to_file=output_file is not None)
+
+    def get_tables(self, only_names=True, verbose=False):
+        """Get the available table in EHST TAP service
+
+        Parameters
+        ----------
+        only_names : bool, TAP+ only, optional, default 'False'
+            True to load table names only
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A list of tables
+        """
+
+        tables = self._tap.load_tables(only_names=only_names,
+                                       include_shared_tables=False,
+                                       verbose=verbose)
+        if only_names is True:
+            table_names = []
+            for t in tables:
+                table_names.append(t.name)
+            return table_names
+        else:
+            return tables
+
+    def get_columns(self, table_name=None, only_names=True, verbose=False):
+        """Get the available columns for a table in EHST TAP service
+
+        Parameters
+        ----------
+        table_name : string, mandatory, default None
+            table name of which, columns will be returned
+        only_names : bool, TAP+ only, optional, default 'False'
+            True to load table names only
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A list of columns
+        """
+
+        if table_name is None:
+            raise ValueError("table name must be specified")
+
+        tables = self._tap.load_tables(only_names=False,
+                                       include_shared_tables=False,
+                                       verbose=verbose)
+        columns = None
+        for t in tables:
+            if str(t.name) == str(table_name):
+                columns = t.columns
+                break
+
+        if columns is None:
+            raise ValueError("table name specified is not found in "
+                             "EHST TAP service")
+
+        if only_names is True:
+            column_names = []
+            for c in columns:
+                column_names.append(c.name)
+            return column_names
+        else:
+            return columns
+
+    def _checkQuantityInput(self, value, msg):
+        if not (isinstance(value, str) or isinstance(value, units.Quantity)):
+            raise ValueError(str(msg) + ""
+                             " must be either a string or astropy.coordinates")
+
+    def _getQuantityInput(self, value, msg):
```

Comment:
the logic in this one is a bit convoluted.  It would be simpler to restructure as:

```
if value is None:
     error msg
elif isinstance(value, str):
    ....
elif isinstance(value, Quantity):
    ...
else:
    error msg
```